feat(Answer: 1): added solution for the Projection problem

Daes-projet opened the pull request:
## Checklist for challenge submission

- [ ] Start your PR message with Answer:${challenge_number}

tomalaforge commented on apps/angular/projection/src/app/ui/card/card.component.ts:
```diff
@@ -1,34 +1,27 @@
-  @Input() type!: CardType;
-  @Input() customClass = '';
+export class CardComponent<T> {
+  @Input({ required: true }) list: T[] | null = null;
```
if it's required, why do you set it to null, prefer initializing list to [] and you can use the empty block of the new @ngfor

tomalaforge commented on apps/angular/projection/src/app/component/student-card/student-card.component.ts:
```diff
@@ -1,35 +1,55 @@
-  cardType = CardType.STUDENT;
+export class StudentCardComponent {
+  private readonly _store = inject(StudentStore);
+  protected readonly students$: Observable<Student[]> = this._store.students$;
```
you can now use signals;

tomalaforge commented:
Good work, I added a few comments.

Daes-projet commented:
> Good work, I added a few comments.

Thank you for reviewing my code and providing comments on areas for improvement. I've replaced magic strings with directives, and I've experimented with using signals instead of RxJS.

tomalaforge commented on apps/angular/projection/src/app/component/city-card/city-card.component.ts:
```diff
@@ -4,30 +4,28 @@ import { CityStore } from '../../data-access/city.store';
-            (deleteEmitter)="deleteCity(city.id)">
-          </app-list-item>
-        </ng-template>
+      <ng-template let-city appCardContent>
```
👍
